pkg/app/piped/driftdetector/detector.go

```diff
@@ -111,6 +114,24 @@ func NewDetector(
 				logger,
 			))
 
+		case model.CloudProviderCloudRun:
+			sg, ok := stateGetter.CloudRunGetter(cp.Name)
+			if !ok {
+				d.logger.Error(fmt.Sprintf(format, cp.Name))
+				continue
```

Comment:
This is obviously the case where is unexpected; how about returning an error on startup as this error only happens with Piped bugs? (Kubernetes as well)

```suggestion
				return fmt.Errorf(format, cp.Name)
```

Reply:
I feel ya.
But I thought it would be a bit strange to make it dependent on other providers.
For instance, driftdetection for Kubernetes will also be affected by the livestatestore for CloudRun bug if we fix the like above

Reply:
I see. Actually both `CloudRunGetter` and `KubernetesGetter` are method of `livestatestore`. So I feel like it's fine to depend on each other. But it's nitpicks, so it's on you.

Reply:
Thank you for your valuable ideas.

@nghialv @khanhtc1202 
If guys have time, I'd really appreciate it if you could give me your opinion.

Reply:
I think it is better to return an error and stop Piped immediately.
But it could be improved by another PR. So adding a TODO is fine at this time.

Reply:
Agree with the stop immediately since there is no mean for piped to keep running without its cloudprovider work.

Reply:
Thank you for everyone's feedback!
I'd like to add the TODO annotation temporarily and fix it in another PR.
TYVM:)

Reply:
👍